commands/template_pull_stack.go

```diff
@@ -0,0 +1,111 @@
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/openfaas/faas-cli/stack"
+	"github.com/spf13/cobra"
+)
+
+var (
+	templateURL    string
+	customRepoName string
+)
+
+func init() {
+	templatePullStackCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
+	templatePullStackCmd.Flags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
+	templatePullStackCmd.PersistentFlags().StringVarP(&customRepoName, "repo", "r", "", "The custom name of the template repo")
+
+	templatePullCmd.AddCommand(templatePullStackCmd)
+}
+
+var templatePullStackCmd = &cobra.Command{
+	Use:   `stack`,
+	Short: `Downloads templates specified in the function definition yaml file`,
+	Long: `Downloads templates specified in the function yaml file, in the current directory
+	`,
+	Example: `
+  faas-cli template pull stack
+  faas-cli template pull stack -f myfunction.yml
+  faas-cli template pull stack -r custom_repo_name
```

Comment:
@martindekov

I get the option 1 uses stack.yml, option 2 uses a specific YAML file, but what does the final option do?

Reply:
if you have listed multiple repos in the yaml you can specify which one to be pulled

Reply:
It seems like scope-creep, but I might have missed something. I think I'm ok with everything now but not convinced about this last feature.

Reply:
So we always want to pull everything in the file? If that is the case there is no problem to remove it